forc/src/cli/commands/plugins.rs

```diff
@@ -18,15 +18,32 @@ pub struct Command {
     describe: bool,
 }
 
+fn get_file_name(path: &Path) -> String {
+    if let Some(Some(path_str)) = path.file_name().map(|path_str| path_str.to_str()) {
+        path_str.to_owned()
+    } else {
+        panic!("Failed to read file name from {path:?}")
+    }
+}
+
 pub(crate) fn exec(command: PluginsCommand) -> ForcResult<()> {
     let PluginsCommand {
         print_full_path,
         describe,
     } = command;
 
+    let mut plugins = crate::cli::plugin::find_all()
+        .map(|path| {
+            print_plugin(path.clone(), print_full_path, describe)
+                .map(|info| (get_file_name(&path), info))
+        })
+        .collect::<Result<Vec<_>, _>>()?;
+    plugins.sort_by(|a, b| a.0.cmp(&b.0));
+    plugins.dedup_by(|a, b| a.0 == b.0);
```

Comment:
How are we deduping here? It should be by what's first in the PATH, or the output of `which {plugin_name}`, right?

Reply:
Perhaps we should put this logic inside of `find_all()`

Reply:
I don't think we have to worry about the version. Fuelup handles that for you by linking your active toolchain binaries to an executable in `~/.fuelup/bin/`. So whatever is linked there is the right version.